### PR TITLE
[Strings] Fix precomputing of StringEq

### PR DESCRIPTION
**Pull request by kripken**
We incorrectly overrode the string operations in the interpreter's subclasses. But
string operations can be implemented in the topmost class there (as they depend on
no module state), so just implement them there, once, in a proper way.

### CODE REVIEW

**Comment by tlively** [comment]
I'm confused; it looks like this PR changes all the string ops _except_ StringEq? How could this have fixed behavior for StringEq?

**Comment by kripken** [comment]
It moves them into the right place. By removing the bad `stringEq` override in the subclass, the good code in the parent now works properly.

File: src/wasm-interpreter.h [hunk 1]
```diff
@@ -2369,31 +2379,6 @@ class ConstantExpressionRunner : public ExpressionRunner<SubType> {
     NOTE_ENTER("Rethrow");
     return Flow(NONCONSTANT_FLOW);
   }
-  Flow visitStringMeasure(StringMeasure* curr) {
-    return Flow(NONCONSTANT_FLOW);
-  }
-  Flow visitStringEncode(StringEncode* curr) { return Flow(NONCONSTANT_FLOW); }
-  Flow visitStringConcat(StringConcat* curr) { return Flow(NONCONSTANT_FLOW); }
-  Flow visitStringEq(StringEq* curr) { return Flow(NONCONSTANT_FLOW); }
```
Comment (kripken) [inline src/wasm-interpreter.h:2377]: This is the critical line for `string.eq` - this line overrode the parent, so we always returned "give up"...

Comment (tlively) [reply src/wasm-interpreter.h:2377]: ohhh got it.

**Review by tlively — APPROVED** [review]

**Pull request merged**